### ISSUE TITLE
target with different aspect ratio

### ISSUE THREAD

**Issue by gzhengli** [AlexeyAB/darknet#1493]
My detection targets have different aspect ratios, and the gap is relatively large. Does this have an impact on training? Can I get the training results?
I used yolov2-tiny.And the part of the output parameters of Training process as follows:

Region Avg IOU: 0.592748, Class: 0.499516, Obj: 0.499963, No Obj: 0.499830, Avg Recall: 1.000000,  count: 3
Region Avg IOU: 0.539716, Class: 0.499853, Obj: 0.499853, No Obj: 0.499830, Avg Recall: 0.666667,  count: 3
Region Avg IOU: 0.473441, Class: 0.499845, Obj: 0.499577, No Obj: 0.499832, Avg Recall: 0.666667,  count: 3
Region Avg IOU: 0.465660, Class: 0.500417, Obj: 0.499944, No Obj: 0.499830, Avg Recall: 0.333333,  count: 3
601: 247.692581, 247.692581 avg, 0.000100 rate, 540.756879 seconds, 57696 images
Loaded: 0.000052 seconds

What is the possible cause of this result?

**Comment by gzhengli** [comment]
The obj and no-obj are all around 0.5 ahd the loss more than 200.
I adjusted the batch ,subdivisions and  learning rate.The lowest loss is 154 and has been oscillating between 200 and 450.

**Comment by AlexeyAB** [comment]
@gzhengli 
* Do you use batch=64 subdivisions=2?
* How many classes and images do you have?
* What mAP can you get after 5000 iterations?
* Try to recalculate anchors, what anchors can you get?
* Try to use `yolov3-tiny.cfg` https://github.com/AlexeyAB/darknet#how-to-train-tiny-yolo-to-detect-your-custom-objects

**Comment by gzhengli** [comment]
AlexeyAB,Thanks for response.

 I have used batch=64  subdivisons=2 and 8,I have tried both cases but have no major changes to the loss,obj and no-obj.
I have two classes and  each category has 500 images for a total of 1000.
Because the computer does not have a GPU, it took a week to train it for more than a thousand times. The loss was always high. I stopped training and didn't achieve  5,000 iterations.
I am using the anchors in yolov2-tiny-voc.cgf, have not changed.It was anchors = 1.08,1.19,  3.42,4.41,  6.63,11.38,  9.42,5.11,  16.62,10.52.
Do you mean that I need to determine the anchors based on the objects I want to detect? How is this calculated?

**Comment by AlexeyAB** [comment]
@gzhengli I didn't check is there any issues or bugs in Training on CPU, since it will take all your live to train it on CPU.

* What pre-trained weights do you use for training?
* Yes, try to use  `yolov3-tiny.cfg`, also recalculate anchors and set it in the 2 [yolo]-layers in the `yolov3-tiny.cfg`
`darknet.exe detector calc_anchors data/obj.data -num_of_clusters 6 -width 416 -height 416`

**Comment by AlexeyAB** [comment]
@gzhengli You should use `yolov3-tiny.conv.15` file as described here: https://github.com/AlexeyAB/darknet#how-to-train-tiny-yolo-to-detect-your-custom-objects

**Comment by gzhengli** [comment]
@AlexeyAB Ok,thank you,I will give it a try.
